docs/source/basic/cheatsheet.rst

```diff
@@ -115,9 +115,10 @@ Create a CPU device for memory allocation on the host side
 
 Allocate a buffer in host memory
   .. code-block:: c++
-
-     Vec<Dim, Idx> extent = value;
-     using BufHost = Buf<DevHost, DataType, Dim, Idx>;
+     // Use alpaka vector as a static array for the extent(dimensions)
```

Comment:
```suggestion
     // Use alpaka vector as a static array for the extent(dimensions)
```
I prefer the word extents over dimensions. In my mental model, if you have a 2D array of size 10x12 elements, it has 2 dimensions and extents 10 and 12. So dimensions for me, are the number of extents.

Reply:
No. When we say "dimensions of an A4 paper" or "dimensions of a box" we don't mean "2 dimensional" or "3 dimentional" we just mean height,width and depth values.

Reply:
Well, an A4 paper has 2 dimensions, and those dimensions are 29.7 cm and 21 cm.

To be pedantic, one could say that an A4 paper has a dimensionality of 2, and its size along those dimensions are 29.7 cm and 21 cm.

In any case, I don't know if there is an official nomenclature in C/C++.
The STL has traits for them, and uses [rank](https://en.cppreference.com/w/cpp/types/rank) for the number of dimensions, and [extents](https://en.cppreference.com/w/cpp/types/extent) the size along each dimension.

Reply:
_Extent_ and _pitch_ are Cuda concepts, C++ developers know "dimensions", "padding", "alignment".  Therefore I have just wanted to add them to clarify.

Reply:
Ok I will do "dimensions array" :)

Reply:
In addition to what @fwyzard said, there is also [`std::extents`](https://en.cppreference.com/w/cpp/container/mdspan/extents).

Reply:
since C++23 ... I still live in C++17 :-)

Reply:
1. I think you mix up "number of dimensions" and the "dimensions". 

2. I haven't replaced or proposed replacing the term extent with the
dimensions, just tried to make the document more understandable for the
people who are new to alpaka.

Reply:
While this mostly counts for me as well, as library designers or architects, we need to stay current with the technical evolution and design accordingly. I also want my libraries to fit the future ;)

Reply:
Alpaka uses of Dim and TDim, which are misleading I think. For example Dim can be NDim ( e.g. matlab or python function names to get number of dimensions)

Reply:
Lets agree on this. Number of Dimensions = Number of Extents. Dimensions could be misleading as you said. I accept. Why, because it could mean a set of dimension names like {XDimension,YDimension,ZDimension} or it could mean as in daily usage {width,depth, height}.  I remove it.

Reply:
> Number of Dimensions = Number of Extents.

Agreed.
It is also called "rank" (in C++) or "dimensionality".